go.mod

```diff
@@ -1,9 +1,15 @@
 module github.com/go-jose/go-jose/v3
 
-go 1.12
+go 1.20
```

Comment:
I think we should leave this as `1.16`. This library is not yet dependent on language features in Go versions >= `1.20`.

The docs for the `go` directive indicate that we should bump the version only when that changes:

> - For packages within the module, the compiler rejects use of language features introduced after the version specified by the go directive. For example, if a module has the directive go 1.12, its packages may not use numeric literals like 1_000_000, which were introduced in Go 1.13.
> - If an older Go version builds one of the module’s packages and encounters a compile error, the error notes that the module was written for a newer Go version. For example, suppose a module has go 1.13 and a package uses the numeric literal 1_000_000. If that package is built with Go 1.12, the compiler notes that the code is written for Go 1.13.

See: https://go.dev/ref/mod#go-mod-file-go